File: functions/functions2/ex12.py
"""Write a function that checks whether a passed string is palindrome or not"""

def is_palindrome(string):
    key = 0
    start_point = 0
    end_point = len(string)-1

    while string[start_point] == string[end_point]:
        if start_point == end_point:
            print("Is palindrome")
            key = 1
            break
        start_point += 1
        end_point -= 1
    if key == 0:
        print("Is not palindrome")

#---------------------------
string = input("Insert a string: ")

is_palindrome(string)

File: functions/functions2/ex12.c
/* Write a function that checks whether a passed string is palindrome or not */

#include <stdio.h>
#include <string.h>

/*say if string is palindromic*/
void is_palindromic(char const *string);

int main()
{
    char string[40];

    printf("Insert a string: ");
    fflush(stdin);
    fgets(string,40,stdin);
    string[strlen(string) - 1] = '\0';

    is_palindromic(string);

    return 0;
}

void is_palindromic(char const *string)
{   
    int key = 0;
    int start_p = 0;
    int end_p = strlen(string) - 1;

    while (string[start_p] == string[end_p])
    {
        if(start_p == end_p) 
        {
            printf("Is palindromic\n");
            key = 1;
            break;
        }
    
        start_p++;
        end_p--;
    }
    if(key == 0) printf("Is not palindromic\n");
}